### ISSUE TITLE
(cdk init): (Does't use pnpm to install dependencies when invoked with pnpx)

### ISSUE THREAD

**Issue by ShivamJoker** [aws/aws-cdk#23205]
### Describe the bug

When I try to init CDK with `pnpm` it starts installing dependencies using `npm`

![screenshot showing npm install getting executed](https://user-images.githubusercontent.com/23727670/205258391-3f2783dc-1c37-4314-aade-f21651123beb.png)

### Expected Behavior

CDK should use `pnpm` to install dependencies instead of just going for `npm`

### Current Behavior

Installs package using npm

### Reproduction Steps

Create a new project with 

```sh
pnpx cdk init app --language typescript
```

### CDK CLI Version

2.53.0

### Node.js Version

16

### OS

Arch Linux

### Language

Typescript

**Comment by peterwoodworth** [comment]
Under the hood we're executing an npm install while `cdk init` command is being ran. We could potentially allow for `pnpm` as well https://github.com/aws/aws-cdk/blob/482dcca501b42d4a1f537274e47193f4db6768ef/packages/aws-cdk/lib/init.ts#L300

**Comment by ShivamJoker** [comment]
I see, I am up for contribution.
How do we know which package manager was used to initiate the install?

Does it get passed in command line arguments? `process.args`

**Comment by peterwoodworth** [comment]
Glad to hear you're interested in contributing! Be sure to check out our [contributing guide](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md)

I'm not sure there's a way to detect if `cdk init` was called with `pnpx`. We may have to introduce a cli option for this

**Comment by mrgrain** [comment]
@ShivamJoker The templates are currently fixed for `npm`. Not that we would never want to support anything else, but they are deliberately basic. 

If you are looking at improving your personal/company workflow. May I suggest you have a look at [projen](https://github.com/projen/projen) which fully support pnpm

**Comment by ShivamJoker** [comment]
> The templates are currently fixed for npm

@mrgrain I don't see the downside of improving the developer experience.
Why to force people to use `npm` if there are better tools available.

I looked at `projen` but it seems like a 3rd party tool which should not be recommended for several reasons like if its updated properly, if it has bugs or security flaws and if we encounter any issues then AWS won't take any responsibility.

**Comment by tripplilley** [comment]
> The templates are currently fixed for `npm`

@mrgrain The [Yarn Rulebook](https://yarnpkg.com/advanced/rulebook) advises [using `$npm_execpath`](https://yarnpkg.com/advanced/rulebook#published-packages-should-avoid-using-npm-run-in-their-scripts) instead of `npm run` in `package.json` scripts. That might help?

It's at least portable amongst `npm`, `pnpm`, and `yarn` (all I've tested so far).

**Comment by mrgrain** [comment]
Absolutely. I reckon there are 3 places that would need to be adjusted:
- the tool that is used to install dependencies
- the scripts in `package.json`
- the `app` in `cdk.json`
